shared/js/wifi_helper.js

```diff
@@ -188,7 +188,10 @@ var WifiHelper = {
       // use ssid + security as a composited key
       var key = network.ssid + '+' +
         self.getSecurity(network).join('+');
-      networksObject[key] = network;
+      if (!networksObject[key] ||
```

Comment:
and please add one more test case on `sharedtest/test/unit/wifi_helper_test` to make sure `networksObject` would be changed if we have another same-name AP with stronger signal.